app/scripts/modules/core/serverGroup/serverGroup.component.ts

```diff
@@ -0,0 +1,149 @@
+import {has} from 'lodash';
+import {IComponentController, IComponentOptions, IFilterService, ITimeoutService, IRootScopeService, IScope, module} from 'angular';
+
+import {SERVER_GROUP_SEQUENCE_FILTER} from 'core/cluster/serverGroup.sequence.filter';
+import {CLUSTER_FILTER_SERVICE} from 'core/cluster/filter/clusterFilter.service';
+import {ENTITY_UI_TAGS_COMPONENT} from 'core/entityTag/entityUiTags.component';
+import {Instance} from 'core/domain/instance';
+import {ServerGroup} from 'core/domain/serverGroup';
+
+interface JenkinsViewModel {
+  number: number;
+  href?: string;
+}
+
+interface ViewModel {
+  waypoint: string;
+  serverGroup: ServerGroup;
+  serverGroupSequence: string[];
+  jenkins: JenkinsViewModel;
+  hasBuildInfo: boolean;
+  instances: Instance[];
+  images?: string;
+}
+
+export class ServerGroupController implements IComponentController {
+  public cluster: string;
+  public serverGroup: ServerGroup;
+  public application: string;
+  public parentHeading: string;
+  public hasLoadBalancers: boolean;
+  public hasDiscovery: boolean;
+
+  public headerIsSticky: () => boolean;
+  public viewModel: ViewModel;
+
+  private lastStringVal: string = null;
+
+  static get $inject(): string[] { return ['$rootScope', '$scope', '$timeout', '$filter', 'clusterFilterService', 'MultiselectModel', 'ClusterFilterModel', 'serverGroupTransformer']; }
+
+  constructor(public $rootScope: IRootScopeService,
+              private $scope: IScope,
+              private $timeout: ITimeoutService,
+              private $filter: IFilterService,
+              private clusterFilterService: any,
+              private MultiselectModel: any,
+              private ClusterFilterModel: any, private serverGroupTransformer: any) {}
+
+  public $onInit(): void {
+    // TODO: Remove $scope. Keeping it here so we can use $watch for now.
+    //       Eventually, there should be events fired when filters change.
+    this.$scope.sortFilter = this.ClusterFilterModel.sortFilter;
+    this.$scope.$watch('sortFilter', () => this.setViewModel(), true);
```

Comment:
You can do this via something like:
```javascript
this.$scope.$watch(() => this.ClusterFilterModel.sortFilter, () => this.setViewModel(), true);
```
which keeps the `sortFilter` off the `$scope`